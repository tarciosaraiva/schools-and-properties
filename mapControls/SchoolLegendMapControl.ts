// @ts-ignore
import SchoolLegendViewMapControl from '~/components/school/SchoolLegendViewMapControl.vue'

export default class SchoolLegendMapControl {
  _container: HTMLDivElement | null = null
  _map: any
  _vm: any

  constructor () {
    this._vm = new SchoolLegendViewMapControl()
  }

  onAdd (map: any) {
    this._map = map
    this._container = document.createElement('div')
    this._container.id = 'schools-legend-view'
    this._container.className = 'mapboxgl-ctrl schools-legend-view-ctrl'
    this._container.appendChild(this._vm.$mount().$el)
    return this._container;
  }

  onRemove () {
    this._container!.parentNode!.removeChild(this._container!);
    this._map = undefined;
  }
}
