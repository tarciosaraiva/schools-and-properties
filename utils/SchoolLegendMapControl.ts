// @ts-ignore
import SchoolLegendMapControl from '~/components/school/SchoolLegendMapControl.vue'

export default class SchoolFilterMapControl {
  _container: HTMLDivElement | null = null
  _map: any
  _vm: any

  constructor () {
    this._vm = new SchoolLegendMapControl()
  }

  onAdd (map: any) {
    this._map = map
    this._container = document.createElement('div')
    this._container.id = 'schools-legend-view'
    this._container.className = 'mapboxgl-ctrl schools-legend-view-ctrl'
    this._container.appendChild(this._vm.$mount().$el)
    return this._container;
  }

  onRemove () {
    this._container!.parentNode!.removeChild(this._container!);
    this._map = undefined;
  }
}
