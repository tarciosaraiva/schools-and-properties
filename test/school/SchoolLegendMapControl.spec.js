import { mount, createLocalVue } from '@vue/test-utils'
import { library } from '@fortawesome/fontawesome-svg-core'
import { fas } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'

import SchoolLegendMapControl from '~/components/school/SchoolLegendMapControl.vue'

library.add(fas)

const localVue = createLocalVue()
localVue.component('FontAwesomeIcon', FontAwesomeIcon)

describe('SchoolLegendMapControl', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(SchoolLegendMapControl, { localVue })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly when collapsed', () => {
    const wrapper = mount(SchoolLegendMapControl, { localVue })
    expect(wrapper).toMatchSnapshot()
  })

  test('renders correctly when expanded', async () => {
    const wrapper = mount(SchoolLegendMapControl, { localVue })
    await wrapper.setData({ collapsed: false })
    expect(wrapper).toMatchSnapshot()
  })

  test('should expand when clicking on caret button', () => {
    const wrapper = mount(SchoolLegendMapControl, { localVue })
    const button = wrapper.find('a')
    button.trigger('click')
    expect(wrapper.vm.collapsed).toBeFalsy()
  })
})
