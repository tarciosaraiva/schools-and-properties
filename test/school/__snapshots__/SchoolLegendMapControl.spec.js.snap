// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolLegendMapControl renders correctly when collapsed 1`] = `
<div class="school-legend-view">
  <div class="d-flex align-items-center justify-content-between">
    <h6>Legend</h6>
    <a href="#">
      <svg aria-hidden="true" class="svg-inline--fa fa-caret-square-up fa-w-14" data-icon="caret-square-up" data-prefix="fas" focusable="false" role="img" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
        <path class="" d="M0 432V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48zm355.515-140.485l-123.03-123.03c-4.686-4.686-12.284-4.686-16.971 0L92.485 291.515c-7.56 7.56-2.206 20.485 8.485 20.485h246.059c10.691 0 16.045-12.926 8.486-20.485z" fill="currentColor" />
      </svg>
    </a>
  </div>
  <div class="collapsable">
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Primary school unrated" src="~/assets/images/primary-school-unrated-small.png">
      <small>Primary school unrated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Primary school unrated" src="~/assets/images/primary-school-rated-small.png">
      <small>Primary school rated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Secondary school unrated" src="~/assets/images/secondary-school-unrated-small.png">
      <small>Secondary school unrated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Secondary school unrated" src="~/assets/images/secondary-school-rated-small.png">
      <small>Secondary school rated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Property marker" src="~/assets/images/property-location-small.png">
      <small>Property marker</small>
    </div>
  </div>
</div>
`;

exports[`SchoolLegendMapControl renders correctly when expanded 1`] = `
<div class="school-legend-view">
  <div class="d-flex align-items-center justify-content-between">
    <h6>Legend</h6>
    <a href="#">
      <svg aria-hidden="true" class="svg-inline--fa fa-caret-square-down fa-w-14" data-icon="caret-square-down" data-prefix="fas" focusable="false" role="img" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
        <path class="" d="M448 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zM92.5 220.5l123 123c4.7 4.7 12.3 4.7 17 0l123-123c7.6-7.6 2.2-20.5-8.5-20.5H101c-10.7 0-16.1 12.9-8.5 20.5z" fill="currentColor" />
      </svg>
    </a>
  </div>
  <div class="collapsable expanded">
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Primary school unrated" src="~/assets/images/primary-school-unrated-small.png">
      <small>Primary school unrated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Primary school unrated" src="~/assets/images/primary-school-rated-small.png">
      <small>Primary school rated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Secondary school unrated" src="~/assets/images/secondary-school-unrated-small.png">
      <small>Secondary school unrated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Secondary school unrated" src="~/assets/images/secondary-school-rated-small.png">
      <small>Secondary school rated</small>
    </div>
    <div class="d-flex align-items-center justify-content-between">
      <img alt="Property marker" src="~/assets/images/property-location-small.png">
      <small>Property marker</small>
    </div>
  </div>
</div>
`;
